device_independent_test/handshake.py

```diff
@@ -0,0 +1,40 @@
+from . import quantum_communicator
+from . import incompatible_measurement
+from . import new_entanglement
+
+class HandShake():
+
+    def __init__(self,communicator: quantum_communicator.QuantumDispatcher):
+        self.dispatcher = communicator
+
+    def dimensionality(self,tolerance,shots):
+        pass
+    def measurement_incompatibility(self,tolerance,shots):
+        result = incompatible_measurement.run_test(self.dispatcher,tolerance,shots)
+        if result[0]:
+            print("Passed Measurment Incompatibility with violation: ", result[1])
+        else:
+            print("Failed Measurment Incompatibility with violation: ", result[1])
+        return result
+    def entanglement(self,tolerance,shots):
+        result = new_entanglement.run_test(self.dispatcher,tolerance,shots)
+        if result[0]:
+            print("Passed Entanglement with violation: ", result[1])
+        else:
+            print("Failed Entanglement with violation: ", result[1])
+        return result
+    def test_all(self,tolerance,shots):
+        pass
+
+    def test(self,name,tolerance,shots):
```

Comment:
I think this method is not really necessary. We can rename `entanglement` to `test_entanglement`, and the user can use them directly. I don't think it's a good practice to use `getattr` if possible.

Reply:
Removed